src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/ImportMetadataFunctionTests.cs

```diff
@@ -241,6 +386,142 @@ public async Task DataSetVersionMeta_CorrectFilters(ProcessorTestData testData,
+
+            var option1Mapping = new FilterOptionMapping
+            {
+                PublicId = "option-1-public-id",
+                Type = MappingType.AutoMapped,
+                CandidateKey = MappingKeyFunctions.FilterOptionKeyGenerator(mappedOption1)
+            };
+
+            var option2Mapping = new FilterOptionMapping
+            {
+                PublicId = "option-2-public-id",
+                Type = MappingType.ManualMapped,
+                CandidateKey = MappingKeyFunctions.FilterOptionKeyGenerator(mappedOption2)
+            };
+
+            var random = new Random();
+
+            var mappings = new DataSetVersionMapping
+            {
+                SourceDataSetVersionId = sourceDataSetVersion.Id,
+                TargetDataSetVersionId = targetDataSetVersion.Id,
+                LocationMappingPlan = new LocationMappingPlan(),
+                FilterMappingPlan = new FilterMappingPlan
+                {
+                    Mappings = testData
+                        .ExpectedFilters
+                        .ToDictionary(
+                            keySelector: MappingKeyFunctions.FilterKeyGenerator,
+                            elementSelector: filter => new FilterMapping
+                            {
+                                OptionMappings = filter
+                                    .Options
+                                    .ToDictionary(
+                                        keySelector: MappingKeyFunctions.FilterOptionKeyGenerator,
+                                        elementSelector: option =>
+                                            option == mappedOption1 ? option1Mapping
+                                            : option == mappedOption2 ? option2Mapping
+                                            : new FilterOptionMapping
+                                            {
+                                                Type = random.Next(1) == 0
+                                                    ? MappingType.AutoNone
+                                                    : MappingType.ManualNone
+                                            })
+                            })
+                }
+            };
+
+            await AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersionMappings.Add(mappings);
+            });
+
+            await ImportMetadata(testData, targetDataSetVersion, instanceId, metaInsertBatchSize);
+
+            await using var publicDataDbContext = GetDbContext<PublicDataDbContext>();
+
+            var actualFilters = await publicDataDbContext.FilterMetas
+                .Include(fm => fm.Options.OrderBy(o => o.Label))
+                .ThenInclude(fom => fom.MetaLinks)
+                .Where(fm => fm.DataSetVersionId == targetDataSetVersion.Id)
+                .OrderBy(fm => fm.Label)
+                .ToListAsync();
+
+            var globalOptionIndex = 0;
```

Comment:
Just wondering, I think I'd actually expect this test to fail... Shouldn't this start counting from the number of options that are already in the database from data set version 1.0 ? 

Or am I not thinking through this correctly?

Reply:
It goes off the starting value of the sequence that is used to control this.  We don't actually set up any prior standing data, so this sequence starts at 0 at this point.